internal/provider/provider.go

```diff
@@ -68,6 +68,7 @@ func configure(ctx context.Context, p *schema.Provider, d *schema.ResourceData)
 		c        = p.Meta().(*scylla.Client)
 	)
 
+	c.Endpoint = defaultEndpoint
 	if endpoint != "" {
```

Comment:
Let's add proper defaults to the provider schema and let's remove this condition.